packages/sonarwhal/src/lib/cli/actions.ts

```diff
@@ -1,13 +1,29 @@
 /**
  * @fileoverview Exports all the actions the CLI is capable of doing.
  */
-import { CLIOptions } from '../types'; // eslint-disable-line no-unused-vars
-import { newRule } from './wizards/new-rule';
-import { initSonarwhalrc } from './wizards/init';
-import { printHelp } from './help';
-import { printVersion } from './version';
-import { analyze } from './analyze';
-import { newParser } from './wizards/new-parser';
+import { CLIOptions } from '../types';
+
+/** Wrapper to dynamically load the different CLI tasks depending on a condition */
+const action = (pkg: string, condition?: string): (actions: CLIOptions) => Promise<boolean> => {
+    return async (actions: CLIOptions): Promise<boolean> => {
+        if (!condition || actions[condition]) {
+            const { default: task } = await import(pkg);
+
+            // TODO: we shouldn't have to pass actions
```

Comment:
I think this depends on the action. `analyze` needs it. The problem I think that is calling the parameter `actions`, it should be called `options`.